library/common/http/dispatcher.h

```diff
@@ -95,13 +93,19 @@ class Dispatcher : public Logger::Loggable<Logger::Id::http> {
     // An implementation option would be to have drainable header maps, or done callbacks.
     std::vector<HeaderMapPtr> metadata_;
     HeaderMapPtr trailers_;
+
+    std::atomic<bool> local_closed_{};
+    std::atomic<bool> remote_closed_{};
```

Comment:
Shouldn't these be guaranteed to only be touched by a single thread?

Reply:
yep they are actually protected, you are right. No need to make them atomic.